src/request_handler.rs

```diff
@@ -1,23 +1,26 @@
 use std::str;
-use std::sync::{Mutex, MutexGuard};
+use std::sync::Arc;
 
 use hyper::rt::{Future, Stream};
 use hyper::{Body, Method, Request, Response, Uri};
 
+use parking_lot::{RwLock, RwLockWriteGuard};
+
 use crate::components::ComponentManager;
 use crate::error::WorkerError;
 use crate::model::ComponentPath;
-use crate::server::BoxedHyperFuture;
-
-lazy_static! {
-    pub static ref GLOBAL_HANDLER: HttpRequestHandler = HttpRequestHandler::new();
-}
 
 // Warning: This method is somewhat complicated, since it needs to deal with async stuff
-pub fn global_request_entrypoint(req: Request<Body>) -> BoxedHyperFuture {
+// TODO: Consider making this a method on a struct somewhere
```

Comment:
I agree with this, for a future refactor change it to a struct method. This is probably "more OO"